lib/element_filter.rb

```diff
@@ -16,13 +16,15 @@
 # you may find current contact information at www.suse.com
 
 class ElementFilter
-  attr_accessor :path, :matchers
+  attr_accessor :path, :operator, :matchers
 
-  def initialize(path, matchers = nil)
+  def initialize(path, operator, matchers = nil)
     @path = path
+    @operator = operator
     @matchers = []
 
-    raise("Wrong type") if ![NilClass, String, Array].include?(matchers.class)
+    raise("Wrong filter type") if ![NilClass, String, Array].include?(matchers.class)
+    raise("Wrong filter operator '#{@operator}'") if !["!=", "="].include?(@operator)
```

Comment:
Maybe it would be good to introduce constants for the operators, so that they don't have to be literally repeated in the code.